custom_components/mammotion/__init__.py

```diff
@@ -50,22 +52,17 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
             f"Could not find Mammotion lawn mower with address {address}"
         )
 
-    device = MammotionBaseBLEDevice(ble_device)
-
-    coordinator = hass.data[DOMAIN][entry.entry_id] = MammotionDataUpdateCoordinator(
+    coordinator = MammotionDataUpdateCoordinator(
         hass,
-        _LOGGER,
         ble_device,
-        device,
-        entry.unique_id,
-        entry.data.get(CONF_NAME, entry.title),
     )
 
     try:
         await coordinator.async_config_entry_first_refresh()
     except BleakNotFoundError as err:
         raise ConfigEntryNotReady from err
 
+    entry.runtime_data = coordinator
```

Comment:
I've not seen this before, is this new in terms of how to do things in HA?

Reply:
yep pretty new https://developers.home-assistant.io/blog/2024/04/30/store-runtime-data-inside-config-entry/

Reply:
Did you update min ha version in the manifest? I assume this breaks older installs?

Reply:
I did, and it does